wagtail/admin/rich_text/editors/draftail/__init__.py

```diff
@@ -2,6 +2,7 @@
 import warnings
 
 from django.forms import Media, widgets
+from django.urls import reverse
```

Comment:
I have been trying a few things out and I think our best bet is to keep with the `reverse_lazy` usage and instead tackle this at the JSON encoding.

This way we can 'resolve' the lazy function when encoding and have a much more generic/robust solution that even works for lazy translations.

https://docs.python.org/3/library/json.html

```suggestion
from django.urls import reverse, reverse_lazy
from django.utils.translation import gettext_lazy
```

The custom encoder would look something like this (I have tested this), it basically just resolves the callable lazy functions if it's found within the nested object.

This would work for wherever we put `reverse_lazy` or `gettext_lazy` in any Draftail entity/feature options.

Note we do not need to do this for the Telepath adapter js args code because it's parsing of strings does something similar already.

```py
# ... other imports
from wagtail.widget_adapters import WidgetAdapter


class LazyStringEncoder(json.JSONEncoder):
    """
    Add support for lazy strings to the JSON encoder so that URLs and
    translations can be resolved when rendering the widget only.
    """

    lazy_string_types = [type(reverse_lazy("")), type(gettext_lazy(""))]

    def default(self, obj):

        if type(obj) in self.lazy_string_types:
            return str(obj)

        return json.JSONEncoder.default(self, obj)
```

@gasman @thibaudcolas  - let me know if this is an anti-pattern or if there is a better approach to resolving these URLs lazily (or a better type check I could be doing).

Reply:
Sorry I missed this, we could use the built-in `DjangoJSONEncoder` to handle lazy strings. I've made a follow-up PR at #11907.

Reply:
Thanks Sage. I'll remember to ping you also for questions like this next time.